saleor/warehouse/availability.py

```diff
@@ -20,11 +20,12 @@ def _get_quantity_allocated(stocks: StockQuerySet) -> int:
 
 def _get_available_quantity(stocks: StockQuerySet) -> int:
     results = stocks.aggregate(
-        total_quantity=Coalesce(Sum("quantity"), 0),
+        total_quantity=Coalesce(Sum("quantity", distinct=True), 0),
         quantity_allocated=Coalesce(Sum("allocations__quantity_allocated"), 0),
```

Comment:
Don't we need `distinct=True` here as well?

Reply:
We don't need it because adding allocations to stock cause multiplying stocks in queryset(Only stocks are multiplied, the number of allocations are fine).